src/main/osd/osd_elements.c

```diff
@@ -227,17 +227,14 @@ static void osdFormatCoordinate(char *buff, char sym, int32_t val)
     // We show 7 decimals, so we need to use 12 characters:
     // eg: s-180.1234567z   s=symbol, z=zero terminator, decimal separator  between 0 and 1
 
-    static const int coordinateMaxLength = 13;//12 for the number (4 + dot + 7) + 1 for the symbol
-
-    buff[0] = sym;
-    const int32_t integerPart = val / GPS_DEGREES_DIVIDER;
-    const int32_t decimalPart = labs(val % GPS_DEGREES_DIVIDER);
-    const int written = tfp_sprintf(buff + 1, "%d.%07d", integerPart, decimalPart);
-    // pad with blanks to coordinateMaxLength
-    for (int pos = 1 + written; pos < coordinateMaxLength; ++pos) {
-        buff[pos] = SYM_BLANK;
+    int pos = 0;
+    buff[pos++] = sym;
+    if (val < 0) {
```

Comment:
Invert `val` if negative, then simply print positive value

Reply:
I don't understand what is "invert". Can you explain it?

Reply:
```
if (val < 0) {
   buff[pos++] = '-';
   val = -val;
}

Reply:
Understood! You always have a different solution for typical problems!

Reply:
In this case it's divide-and-conquer problem - handle sign and then handle rest (print unsigned fixed-point value). IMO it will help with code readability and maintainability.
It will also make code a bit smaller.

It's up to you to decide which approach is better, just consider it as option

BTW: using better printf will solve this in much cleaner way (`"%.7f") and as a bonus it will handle rounding correctly ...

Reply:
Implemented the suggestion. I like the result. :-)